crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -569,6 +604,205 @@ pub fn if_with_same_arms(checker: &mut Checker, stmt: &Stmt, parent: Option<&Stm
+                                        return true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+fn constant_to_str(value: &Constant) -> String {
+    match value {
+        Constant::None => "None".to_string(),
+        Constant::Str(s) => format!("{:?}", s),
+        Constant::Int(i) => i.to_string(),
+        Constant::Float(f) => f.to_string(),
+        Constant::Ellipsis => "...".to_string(),
+        Constant::Bytes(b) => {
+            let the_string = String::from_utf8(b.clone()).unwrap();
+            format!("b\"{the_string}\"")
+        }
+        Constant::Complex { real, imag } => format!("{}+{}j", real, imag),
+        Constant::Bool(b) => match b {
+            true => "True".to_string(),
+            false => "False".to_string(),
+        },
+        Constant::Tuple(items) => {
+            let mut result = "(".to_string();
+            for item in items {
+                result.push_str(&constant_to_str(item));
+                result.push_str(", ");
+            }
+            result.push(')');
+            result
+        }
+    }
+}
+
+/// SIM116
+pub fn if_to_dict(checker: &mut Checker, stmt: &Stmt, test: &Expr, body: &[Stmt], orelse: &[Stmt]) {
+    if !should_proceed_main(test, body, orelse) {
+        return;
+    }
+    let variable: String;
+    let mut child: Option<&Stmt> = orelse.get(0);
+    let mut else_value: Option<String> = None;
+    let mut key_value_pairs: HashMap<String, String> = HashMap::new();
+    // This check is useless because we also check above, but it makes Rust happy
+    if let ExprKind::Compare {
+        left, comparators, ..
+    } = &test.node
+    {
+        if let ExprKind::Constant { value, .. } = &comparators[0].node {
+            let key = constant_to_str(value);
+            if let StmtKind::Return { value } = &body[0].node {
+                let final_value = match value {
+                    Some(value) => checker
+                        .locator
+                        .slice_source_code_range(&Range::from_located(value)),
+                    None => return,
+                };
+                key_value_pairs.insert(key, final_value.to_string());
+            }
+        }
+        if let ExprKind::Name { id, .. } = &left.node {
+            variable = id.clone();
+        } else {
+            return;
+        }
+    } else {
+        return;
+    }
+    while child.is_some() {
+        if !should_proceed_child(child.unwrap(), &variable) {
+            return;
+        }
+        if let StmtKind::If { test, body, orelse } = &child.unwrap().node {
+            if let StmtKind::Return { value } = &body[0].node {
+                let clean_value = match value {
+                    Some(item) => item,
+                    None => return,
+                };
```

Comment:
Nit: You can use Rust's let else chain to simplify this code
```suggestion
                let Some(clean_value) = value else { return; };
```